OPHD/UI/StringTable.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <NAS2D/Renderer/Renderer.h>
+#include <NAS2D/Renderer/Color.h>
+#include <NAS2D/Renderer/Point.h>
+#include <NAS2D/Renderer/Vector.h>
+#include <string>
+#include <vector>
+#include <cstddef>
+
+// Draw a 2 dimensional table of text. Determine cell size based on inserted text, font, and padding. Only allows one line of text per cell.
+class StringTable
+{
+public:
+	using CellCoordinate = NAS2D::Point<std::size_t>;
+
+	enum class Justification
+	{
+		Left,
+		Right
+	};
+
+	struct Cell
+	{
+		// Set textColor to ColorEmpty to indicate cell should use default StringTable color
+		static const NAS2D::Color ColorEmpty;
+
+		// Use defaultFont if not set
+		NAS2D::Font* font = nullptr;
```

Comment:
My purpose in using nullptr for font and ColorEmpty in the cell is to allow dynamically changing the text and color of the cell after adding data. If you specifically set the text color or font of the cell, but used what was the default value, the cell would retain this value even if the actual default was changed. 

Also, setting default font and text color of a cell does not require pulling the default table values, since you can just use nullptr or ColorEmpty. 

Maybe this behaviour is too complex or subtle though? We could mandate that default color and font are set in the constructor and is considered nonmutable.

Reply:
Ahh, that explains it. Yeah, I guess I would have gone with a non-mutable assumption for the defaults. Would seem odd to be changing those dynamically after construction. Maybe create some sort of disco table.

Reply:
Yeah, could be a disco table. :)